lib/api/core/hotelClerk.js

```diff
@@ -561,26 +451,23 @@ function cleanUpRooms (roomId) {
      */
     this.rooms[roomId].destroyed = true;
 
-    this.kuzzle.dsl.removeRoom(this.rooms[roomId])
+    return this.kuzzle.dsl.remove(roomId)
       .then(() => {
         this.kuzzle.pluginsManager.trigger('room:remove', roomId);
-        deferred.resolve(roomId);
+        return roomId;
```

Comment:
It leads to the same result: using [propagation](https://github.com/kriskowal/q#propagation), a promise chain resolves promises until it gets a value, or until it encounters a throw/promise rejection.

So returning a new promise resolving to `roomId` will only add a promise to the chain, postponing the result with no gain.